homeassistant/components/home_connect/sensor.py

```diff
@@ -182,12 +331,12 @@ async def async_update(self) -> None:
 class HomeConnectAlarmSensor(HomeConnectEntity, SensorEntity):
     """Sensor entity setup using SensorEntityDescription."""
 
-    entity_description: HomeConnectSensorEntityDescription
+    entity_description: HomeConnectAlarmSensorEntityDescription
```

Comment:
To me the main reason to use `EntityDescription` data classes was to have a [unified class](https://github.com/beastie29a/core/blob/fb3a4993d040fcf8fc93e92c03a0312f91d68ee2/homeassistant/components/home_connect/sensor.py#L373-L406) for the Platform to make it easier for entities to be added in the future since a developer or user would only need to add an item to the appropriate tuple instead of duplicating a class and changing its logic.

Separate classes also make it similar to the current method of adding entities which, as you've probably seen in the switch entity, can make adding additional entities complicated and add more duplicate code than necessary.

Reply:
I understand your idea, however, I think that simplifying so much the sensor entity class results in more complicated entity descriptions that contains duplicated code.
Also, this PR was about door enum entities, code quailty improvements and simplifications comes in another PR I'm preparing these days.